src/transaction/log_recovery_redo_parallel.hpp

```diff
@@ -337,6 +390,30 @@ namespace cublog
     return NO_ERROR;
   }
 
+  template <typename TYPE_LOG_REC>
+  constexpr short redo_job_replication_delay_impl<TYPE_LOG_REC>::SENTINEL_VOLID;
+  template <typename TYPE_LOG_REC>
+  constexpr int32_t redo_job_replication_delay_impl<TYPE_LOG_REC>::SENTINEL_PAGEID;
+  template <typename TYPE_LOG_REC>
+  constexpr vpid redo_job_replication_delay_impl<TYPE_LOG_REC>::SENTINEL_VPID;
+
+  template <typename TYPE_LOG_REC>
+  redo_job_replication_delay_impl<TYPE_LOG_REC>::redo_job_replication_delay_impl (
+	  const log_lsa &a_rcv_lsa, time_t a_start_time_msec, const char *a_source)
+    : redo_parallel::redo_job_base (SENTINEL_VPID, a_rcv_lsa)
+    , m_start_time_msec (a_start_time_msec)
+    , m_source (a_source)
+  {
+  }
+
+  template <typename TYPE_LOG_REC>
+  int  redo_job_replication_delay_impl<TYPE_LOG_REC>::execute (THREAD_ENTRY *thread_p, log_reader &,
+      LOG_ZIP &, LOG_ZIP &)
+  {
+    const int res = log_rpl_calculate_replication_delay (thread_p, m_start_time_msec, m_source);
+    return res;
+  }
```

Comment:
Move definition to .cpp after removing template.

Reply:
Obviously